File: mobile/lib/models/shift.dart
class Shift {
  final int id;
  final int userId;
  final String startTime;
  final String endTime;
  final String? clockInTime;
  final String? clockOutTime;
  final String status;
  final String? notes;

  Shift({
    required this.id,
    required this.userId,
    required this.startTime,
    required this.endTime,
    this.clockInTime,
    this.clockOutTime,
    required this.status,
    this.notes,
  });

  factory Shift.fromJson(Map<String, dynamic> json) {
    return Shift(
      id: json['ID'],
      userId: json['UserID'],
      startTime: json['StartTime'],
      endTime: json['EndTime'],
      clockInTime: json['ClockInTime'],
      clockOutTime: json['ClockOutTime'],
      status: json['Status'] ?? 'scheduled',
      notes: json['Notes'],
    );
  }
}

File: domain/shift.go
package domain

import (
	"time"

	"gorm.io/gorm"
)

// Shift represents a scheduled work period for a user at a specific location
type Shift struct {
	gorm.Model
	UserID        uint       `gorm:"not null;index"`
	LocationID    uint       `gorm:"not null;index"`
	TaskID        *uint      `gorm:"index"`
	StartTime     time.Time  `gorm:"not null;index"`
	EndTime       time.Time  `gorm:"not null"`
	ClockInTime   *time.Time `gorm:"index"`
	ClockOutTime  *time.Time `gorm:"index"`
	ProofImage    string     `gorm:"type:text"` // Base64 encoded image or path
	Notes         string     `gorm:"type:text"`
	Status        string     `gorm:"type:varchar(20);default:'scheduled'"` // e.g., scheduled, completed, cancelled
}
